src/functional/fencil_processors/source_modules/source_modules.py

```diff
@@ -29,14 +55,14 @@ class ScalarParameter:
 @dataclass(frozen=True)
 class BufferParameter:
     name: str
-    dimensions: Sequence[str]
+    dimensions: tuple[str, ...]
```

Comment:
Changing to tuple here and elsewhere to ensure they stay immutable (hashable), which is essential for caching to work well in the future.